lib/ansible/modules/web_infrastructure/acme_certificate_revoke.py

```diff
@@ -0,0 +1,314 @@
+    if openssl_version < (1, 1):
+        host_header = ['Host', _get_host_from_uri(ocsp_uri)]
+    else:
+        host_header = ['Host=' + _get_host_from_uri(ocsp_uri)]
+    # Compose arguments for OpenSSL
+    args = ['ocsp', '-no_nonce', '-header']
+    args.extend(host_header)
+    args.extend(['-issuer', intermediate_certificate, '-cert', certificate, '-url', ocsp_uri, '-VAfile', intermediate_certificate])
+    result = _run_openssl(module, args)
+
+    # Interpret result
+    m = re.match(r'^.*: ([a-zA-Z]+)(?:\n|\r|$)', result)
+    if not m:
+        m = re.match(r'^Responder Error: (.*)(?:\n|\r|$)', result)
+        if m:
+            raise ModuleFailException('OCSP responder error: {0}'.format(m[1]))
+        raise ModuleFailException('Cannot parse OpenSSL OCSP output: {0}'.format(result))
+
+    return m[1] == 'revoked'
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            account_key_src=dict(type='path', aliases=['account_key']),
+            account_key_content=dict(type='str', no_log=True),
+            acme_directory=dict(required=False, default='https://acme-staging.api.letsencrypt.org/directory', type='str'),
+            acme_version=dict(required=False, default=1, choices=[1, 2], type='int'),
+            validate_certs=dict(required=False, default=True, type='bool'),
+            private_key_src=dict(type='path'),
+            private_key_content=dict(type='str', no_log=True),
+            certificate=dict(required=True, type='path'),
+            intermediate_certificate=dict(required=False, type='path'),
+            revoke_reason=dict(required=False, type='int'),
+            force=dict(required=False, type='bool', default=False),
+        ),
+        required_one_of=(
+            ['account_key_src', 'account_key_content', 'private_key_src', 'private_key_content'],
+        ),
+        mutually_exclusive=(
+            ['account_key_src', 'account_key_content', 'private_key_src', 'private_key_content'],
+        ),
+        required_if=(
+            # Make sure that if force is set to False, intermediate_certificate is specified
+            ['force', False, ['intermediate_certificate'], True],
+        ),
+        supports_check_mode=True,
+    )
+
+    if not module.params.get('validate_certs'):
+        module.warn(warning='Disabling certificate validation for communications with ACME endpoint. ' +
+                            'This should only be done for testing against a local ACME server for ' +
+                            'development purposes, but *never* for production purposes.')
+
+    try:
+        if not module.params['force']:
+            # Check whether certificate is already revoked
+            if is_revoked_ocsp(module, module.params.get('certificate'), module.params.get('intermediate_certificate')):
+                module.exit_json(changed=False)
+
+        # From this point on, we assume the certificate has to be revoked
+        if not module.check_mode:
```

Comment:
hmm it looks like this module does "not really" support check mode, does it? Better use supports_check_mode=False?

Reply:
That's true. I'm currently trying to change this, but I can re-add the check mode later.

Reply:
I've left the `if not module.check_mode:` in, so it is easier to enable it back later, but I've commented out the `supports_check_mode` with a comment explaining why it is currently not there.

Reply:
Ok, I've been thinking some more about this. A check mode will probably never work reliably. I'm even thinking about removing the OCSP check and make `force == True` the default. This will make usage simpler (the intermediate certificate doesn't has to be specified), simplify the code, and make the plugin more reliable (since OCSP isn't very reliable in general).

Reply:
I've moved the OCSP code into a gist (https://gist.github.com/felixfontein/40d817d87a89548e4c227dea80424fa6) in case we want to re-use it somewhere/somewhen later.